packages/fether-electron/src/main/app/messages/index.js

```diff
@@ -45,6 +46,11 @@ export default async (fetherApp, event, data) => {
 
         break;
       }
+      case 'RESTART_NODE_REQUEST': {
+        setupParityEthereum({});
```

Comment:
Well, if `hasCalledInitParityEthereum` was actually set to `true` where it should have been in the code, this would have thrown `Unable to initialise Parity Ethereum more than once` :P

Reply:
Also, `setupParityEthereum` takes a parameter and passes down `.win` to `ParityEthereum`, but the `ParityEthereum` constructor doesn't take any argument; let's fix this while we're at it.

Reply:
> Well, if hasCalledInitParityEthereum was actually set to true where it should have been in the code, this would have thrown Unable to initialise Parity Ethereum more than once :p

We could for example:
- have the current constructor logic of ParityEthereum inside `(new ParityEthereum).init()`
- store ParityEthereum in FetherApp
- use fetherApp.parityEthereum.init() in the message handler

Reply:
> or not have ParityEthereum be a class...

lol, that's what I wanted to find somewhere in the code, but everything seems to be a class.

I just did as if `setupParityEthereum` was that function, didn't think too much about it. I propose just to remove the argument of this function, and not refactor too much (leave it for a next PR refactor PR). What do you think?

Reply:
The return value of `setupParityEthereum` (the instance of ParityEthereum) isn't used anywhere, and ParityEthereum is only used by `setupParityEthereum`. So I think it's safe to move the code of the constructor of ParityEthereum (minus `hasCalledInitParityEthereum`) as the contents of `setupParityEthereum`. I don't think it's too much work, but I don't mind if we do this in another PR